File: server/app/schemas/market.py
"""
行情数据模式
"""

from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class Stock(BaseModel):
    """股票数据"""
    code: str
    name: str
    market: str  # A股、港股、美股等
    price: float
    change: float
    change_percent: float
    volume: float
    amount: float
    high: float
    low: float
    open: float
    pre_close: float
    timestamp: datetime
    
    class Config:
        from_attributes = True


class KLineData(BaseModel):
    """K线数据"""
    date: str
    open: float
    high: float
    low: float
    close: float
    volume: float
    amount: float


class HistoryData(BaseModel):
    """历史数据"""
    date: str
    price: float
    volume: float
    amount: float

File: extension/src/types/market.ts
/**
 * 行情数据类型定义
 */

import { MarketType } from './common';

export interface Stock {
    code: string;
    name: string;
    market: MarketType;
    price: number;
    change: number;
    changePercent: number;
    volume: number;
    amount: number;
    high: number;
    low: number;
    open: number;
    preClose: number;
    timestamp: Date;
}

export interface KLineData {
    date: string;
    open: number;
    high: number;
    low: number;
    close: number;
    volume: number;
    amount: number;
}

export interface HistoryData {
    date: string;
    price: number;
    volume: number;
    amount: number;
}

export interface TechnicalIndicator {
    ma5?: number;
    ma10?: number;
    ma20?: number;
    ma30?: number;
    ma60?: number;
    macd?: {
        dif: number;
        dea: number;
        macd: number;
    };
    kdj?: {
        k: number;
        d: number;
        j: number;
    };
    rsi?: number;
}

